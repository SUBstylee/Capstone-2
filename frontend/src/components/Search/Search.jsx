import {useState} from 'react'
import { Form } from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';
import './Search.scss';

const Search = () => {
    const [active,setActive]=useState(false);
    const [keyword,setKeyword]=useState('');

    const navigate=useNavigate();

    const handleClick=()=>{
        setActive(!active);
    };

    const submitHandler=(e)=>{
        e.preventDefault();
        console.log('submitted');
        if(keyword){
            navigate(`/search/${keyword.trim()}`);
        }else{
            navigate('/');
        };
    };

  return (
    <div className={`Search ${active?'active':''}`}>
        <Form onSubmit={submitHandler}>
            <input type='text' name='q' className='Search-input' placeholder='Search Products...' onChange={(e)=>setKeyword(e.target.value)}/>
        </Form>
        <button className='Search-btn' onClick={handleClick}>
            <i className='fas fa-search'></i>
        </button>
    </div>
  )
}

export default Search