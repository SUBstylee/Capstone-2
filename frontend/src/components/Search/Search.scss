.Search{
    position: relative;
    width: 230px;

    .Search-input,.Search-btn{
        border-radius: 8px;
        margin-left: 3px;
        height: 40px;
        width: 40px;
    }
    .Search-input{
        background-color: #fff;
        border: 0;
        font-size: 18px;
        padding: 15px;
        transition: width 0.3s ease;
    };
    .Search-btn{
        background-color: #fff;
        border: 1px solid #919191;
        color: #919191;
        cursor: pointer;
        font-size: 24px;
        position: absolute;
        top: 0;
        left: 0;
        transition: transform 0.3s ease;
    };
    .Search-btn:focus,.Search-input:focus{
        outline: none;
    };
    &.active .Search-input{
        width: 200px;
    };
    &.active .Search-btn{
        transform: translateX(198px);
    };
};