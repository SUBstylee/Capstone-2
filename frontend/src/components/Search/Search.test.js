import { render } from '@testing-library/react';
import Search from './Search.jsx';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

test("renders Search component without crashing", () => {
    const { queryByPlaceholderText } = render(
        <Provider store={store}><BrowserRouter><Search /></BrowserRouter></Provider>
    );

    expect(queryByPlaceholderText(/Search Products.../i)).toBeInTheDocument();
});

test("matches snapshot", function () {
    const { asFragment } = render(
        <Provider store={store}><BrowserRouter><Search /></BrowserRouter></Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});