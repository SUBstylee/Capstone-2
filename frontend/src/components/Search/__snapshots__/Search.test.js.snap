// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Search "
  >
    <form
      class=""
    >
      <input
        class="Search-input"
        name="q"
        placeholder="Search Products..."
        type="text"
      />
    </form>
    <button
      class="Search-btn"
    >
      <i
        class="fas fa-search"
      />
    </button>
  </div>
</DocumentFragment>
`;
